hypershift-operator/controllers/platform/aws/controller.go

```diff
@@ -148,12 +198,46 @@ func (r *AWSEndpointServiceReconciler) Reconcile(ctx context.Context, req ctrl.R
 	return ctrl.Result{}, nil
 }
 
-func reconcileAWSEndpointService(ctx context.Context, awsEndpointService *hyperv1.AWSEndpointService, ec2Client ec2iface.EC2API, elbv2Client elbv2iface.ELBV2API) error {
+func listNodePools(ctx context.Context, c client.Client, nodePoolNamespace string, clusterName string) ([]hyperv1.NodePool, error) {
+	nodePoolList := &hyperv1.NodePoolList{}
+	if err := c.List(ctx, nodePoolList, &client.ListOptions{Namespace: nodePoolNamespace}); err != nil {
+		return nil, fmt.Errorf("failed to list NodePools in namespace %s for cluster %s : %w", nodePoolNamespace, clusterName, err)
+	}
+	filtered := []hyperv1.NodePool{}
+	for i, nodePool := range nodePoolList.Items {
+		if nodePool.Spec.ClusterName == clusterName {
+			filtered = append(filtered, nodePoolList.Items[i])
+		}
+	}
+	return filtered, nil
+}
+
+func listSubnetIDs(ctx context.Context, c client.Client, clusterName, nodePoolNamespace string) ([]string, error) {
+	nodePools, err := listNodePools(ctx, c, nodePoolNamespace, clusterName)
+	if err != nil {
+		return nil, err
+	}
+	subnetIDs := []string{}
+	for _, nodePool := range nodePools {
+		if nodePool.Spec.Platform.AWS != nil &&
+			nodePool.Spec.Platform.AWS.Subnet != nil &&
+			nodePool.Spec.Platform.AWS.Subnet.ID != nil {
+			subnetIDs = append(subnetIDs, *nodePool.Spec.Platform.AWS.Subnet.ID)
+		}
+	}
+	sort.Strings(subnetIDs)
+	return subnetIDs, nil
+}
+
+func reconcileAWSEndpointService(ctx context.Context, awsEndpointService *hyperv1.AWSEndpointService, subnetIDs []string, ec2Client ec2iface.EC2API, elbv2Client elbv2iface.ELBV2API) error {
 	log, err := logr.FromContext(ctx)
 	if err != nil {
 		return fmt.Errorf("no logger found: %w", err)
 	}
 
+	// reconcile subnetIDs from node pools
+	awsEndpointService.Spec.SubnetIDs = subnetIDs
```

Comment:
Should we update the object in the kube api if this results in a change to improve debuggability? Right now we only update .Status

Reply:
+1

Reply:
Now that you point this out, I'm not exactly sure how the spec is being updated (it is) :thinking:

Reply:
I was testing with an old `hypershift-operator` binary that still had the code in the other controller :man_facepalming: